commit e1619f36bf849ba183ae5fb53f4794d5273ea65c
Author: Ray Strode <rstrode@redhat.com>
Date:   Tue Feb 24 13:36:15 2009 -0500

    Fix heap corruptor in ply-boot-server
    
    This fixes a memory corruption bug that
    cropped up during a round of leak fixes
    a while back.

diff --git a/src/ply-boot-server.c b/src/ply-boot-server.c
index 8615f3b..77115e5 100644
--- a/src/ply-boot-server.c
+++ b/src/ply-boot-server.c
@@ -196,7 +196,7 @@ ply_boot_connection_read_request (ply_boot_connection_t  *connection,
 
       if (!ply_read (connection->fd, &argument_size, sizeof (uint8_t)))
         {
-          free(command);
+          free (*command);
           return false;
         }
 
@@ -204,7 +204,7 @@ ply_boot_connection_read_request (ply_boot_connection_t  *connection,
 
       if (!ply_read (connection->fd, *argument, argument_size))
         {
-          free(command);
+          free (*command);
           return false;
         }
     }
@@ -307,7 +307,7 @@ ply_boot_connection_on_request (ply_boot_connection_t *connection)
                       strlen (PLY_BOOT_PROTOCOL_RESPONSE_TYPE_NAK)))
         ply_error ("could not write bytes: %m");
 
-      free(command);
+      free (command);
       return;
     }
 
